### ISSUE TITLE
Julia component generator should escape $ within docstrings

### ISSUE THREAD

**Issue by rpkyle** [plotly/dash#1269]
Following my attempt to build DashTable.jl, the following error appeared:

```julia
[ Info: Precompiling DashTable [1b08a953-4be3-4667-9a23-759fc69216d1]
ERROR: LoadError: LoadError: syntax: invalid interpolation syntax: "$'"
```

The culprit appears to be two occurrences of `$'` within the docstring, based on my reading of this issue: https://github.com/JuliaLang/julia/issues/10816.

We should modify the component generator to prepend a `\` to escape `$` whenever they occur, which should resolve the error at component build time in the future.

A similar change was made for the R component generator, to address unescaped backslashes:
https://github.com/plotly/dash/blob/316b6628033e1acf2810f95640e64cf3babae913/dash/development/_r_components_generation.py#L411-L413

**Comment by alexcjohnson** [comment]
@waralex included this in #1302 - now part of #1189 -> [here](https://github.com/plotly/dash/pull/1197/files#diff-4de66a02832ac93535450311a2298fdeR449)

I haven't tried building `DashTable.jl` yet but I'm assuming this is fixed.

**Comment by rpkyle** [comment]
> @waralex included this in #1302 - now part of #1189 -> [here](https://github.com/plotly/dash/pull/1197/files#diff-4de66a02832ac93535450311a2298fdeR449)
> 
> I haven't tried building `DashTable.jl` yet but I'm assuming this is fixed.

Yep, looks to have been fixed:

https://github.com/plotly/dash/blob/b5e59a23033de9ce65e5ec67840240b1005ada04/dash/development/_jl_components_generation.py#L436-L442

I'll go ahead and close now.
